libraries/AC_AutoTune/AC_AutoTune_Heli.cpp

```diff
@@ -499,91 +461,95 @@ void AC_AutoTune_Heli::load_intra_test_gains()
+            rate_p = tune_roll_rp;
+            rate_d = tune_roll_rd;
         }
-        attitude_control->get_rate_roll_pid().ff(tune_roll_rff);
-        attitude_control->get_rate_roll_pid().filt_T_hz(orig_roll_fltt);
-        attitude_control->get_rate_roll_pid().slew_limit(0.0f);
-        attitude_control->get_angle_roll_p().kP(tune_roll_sp);
+        load_gain_set(ROLL, rate_p, rate_i, rate_d, tune_roll_rff, tune_roll_sp, tune_roll_accel, orig_roll_fltt, 0.0f, 0.0f);
         break;
     case PITCH:
         if (tune_type == SP_UP) {
-            attitude_control->get_rate_pitch_pid().kI(orig_pitch_ri);
+            rate_i = orig_pitch_ri;
         } else {
             // freeze integrator to hold trim by making i term small during rate controller tuning
-            attitude_control->get_rate_pitch_pid().kI(0.01f * orig_pitch_ri);
+            rate_i = 0.01f * orig_pitch_ri;
         }
         if (tune_type == MAX_GAINS && !is_zero(tune_pitch_rff)) {
-            attitude_control->get_rate_pitch_pid().kP(0.0f);
-            attitude_control->get_rate_pitch_pid().kD(0.0f);
+            rate_p = 0.0f;
+            rate_d = 0.0f;
         } else {
-            attitude_control->get_rate_pitch_pid().kP(tune_pitch_rp);
-            attitude_control->get_rate_pitch_pid().kD(tune_pitch_rd);
+            rate_p = tune_pitch_rp;
+            rate_d = tune_pitch_rd;
         }
-        attitude_control->get_rate_pitch_pid().ff(tune_pitch_rff);
-        attitude_control->get_rate_pitch_pid().filt_T_hz(orig_pitch_fltt);
-        attitude_control->get_rate_pitch_pid().slew_limit(0.0f);
-        attitude_control->get_angle_pitch_p().kP(tune_pitch_sp);
+        load_gain_set(PITCH, rate_p, rate_i, rate_d, tune_pitch_rff, tune_pitch_sp, tune_pitch_accel, orig_pitch_fltt, 0.0f, 0.0f);
         break;
     case YAW:
         // freeze integrator to hold trim by making i term small during rate controller tuning
-        attitude_control->get_rate_yaw_pid().kP(tune_yaw_rp);
-        attitude_control->get_rate_yaw_pid().kI(tune_yaw_rp*0.01f);
-        attitude_control->get_rate_yaw_pid().kD(tune_yaw_rd);
-        attitude_control->get_rate_yaw_pid().ff(tune_yaw_rff);
-        attitude_control->get_rate_yaw_pid().filt_E_hz(tune_yaw_rLPF);
-        attitude_control->get_rate_yaw_pid().filt_T_hz(orig_yaw_fltt);
-        attitude_control->get_rate_yaw_pid().slew_limit(0.0f);
-        attitude_control->get_angle_yaw_p().kP(tune_yaw_sp);
+        load_gain_set(YAW, tune_yaw_rp, tune_yaw_rp*0.01f, tune_yaw_rd, tune_yaw_rff, tune_yaw_sp, tune_yaw_accel, orig_yaw_fltt, tune_yaw_rLPF, 0.0f);
+        break;
+    }
+}
+
+// load gains
+void AC_AutoTune_Heli::load_gain_set(AxisType s_axis, float rate_p, float rate_i, float rate_d, float rate_ff, float angle_p, float max_accel, float rate_fltt, float rate_flte, float smax)
+{
+    switch (axis) {
+    case ROLL:
+        attitude_control->get_rate_roll_pid().kP(rate_p);
```

Comment:
You might save a few bytes here by creating a:
```
AC_PID *pid;


    switch (axis) {
    case ROLL:
       pid = attitude_control->get_rate_roll_pid();
   .
.
.
```

and using that for each case

Reply:
@peterbarker Thanks for the suggestion.    So that get_* function actually sets the gain when the argument is included. The place where your showing that this might be useful is here
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AC_AutoTune/AC_AutoTune_Heli.cpp#L372

Reply:
I'd actually played around with having a structure containing all these gains - so you new method would take a structure rather than half-a-bajillion parameters.  That fell apoart a little on the yaw pid which was slightly different to the others IIRC.